### PR TITLE
Test file permissions for auto-instrumentations

### PR DESCRIPTION
**Pull request by pavolloffay**
Signed-off-by: Pavol Loffay <p.loffay@gmail.com>

Related to #640

### CODE REVIEW

File: tests/e2e/instrumentation-java/01-install-app.yaml [hunk 1]
```diff
@@ -15,6 +15,10 @@ spec:
         sidecar.opentelemetry.io/inject: "true"
         instrumentation.opentelemetry.io/inject-java: "true"
     spec:
+      securityContext:
```
Comment (jpkrohling) [inline tests/e2e/instrumentation-java/01-install-app.yaml:18]: Would this work on OpenShift? My understanding is that it needs a special SCC configuration done by the cluster-admin in order to allow this config at the workload level.

Note also that OpenShift (and likely other platforms) sets a random user ID to the actual processes, so, if you need the user ID to be a specific number, your code will certainly break at runtime.

Comment (pavolloffay) [reply tests/e2e/instrumentation-java/01-install-app.yaml:18]: I don't know if this works on OCP or other platforms. 

The previous commit shows that it fails if the image does not have `go+r`.

**Comment by pavolloffay** [comment]
@jpkrohling any update from your side on this PR? 

The goal of this PR is to add tests for https://github.com/open-telemetry/opentelemetry-operator/pull/640. The tests fail without that PR being merged.

**Review by jpkrohling — APPROVED** [review]
It's OK for me, but if this means we have to rely on SCC to get things to work, then things might *not* work on multi-tenant Kubernetes platforms, as SCC is disabled to regular, non-admin users.

**Pull request merged**